bentoml/_internal/models/model.py

```diff
@@ -318,25 +328,30 @@ def __init__(self, base_path: "t.Union[PathType, FS]"):
 @attr.frozen
 class ModelContext:
     framework_name: str
-    framework_versions: dict[str, str]
-    bentoml_version: str = BENTOML_VERSION
-    python_version: str = PYTHON_VERSION
+    framework_versions: t.Dict[str, str]
+    bentoml_version: str = attr.field(default=BENTOML_VERSION)
+    python_version: str = attr.field(default=PYTHON_VERSION)
 
     @staticmethod
-    def from_dict(data: dict[str, str | dict[str, str]]) -> ModelContext:
-        return cattrs.structure(data, ModelContext)
+    def from_dict(data: dict[str, str | dict[str, str]] | ModelContext) -> ModelContext:
+        if isinstance(data, ModelContext):
+            return data
+        return bentoml_cattr.structure(data, ModelContext)
+
+attr.resolve_types(ModelContext, globals(), locals())
```

Comment:
@sauyon looks like this is required for cattrs to work, when using `from __future__ import annotations`, see https://www.attrs.org/en/stable/types.html

Reply:
Ah, got it! I think according to the cattrs people `cattrs.Converter()` should also be able to structure these types.

Reply:
Though probably for our types we want to do this anyway.

Reply:
@sauyon cattrs.Convertor doesn't have a resolve_types method, it seems to be a attrs behavior that cattrs did not expect